### PR TITLE
[DRAFT] arm64: dts: qcom: msm8916-samsung-gt58lte: add initial dts for Galaxy 

### PR DESCRIPTION
**Pull request by hello-world-321**
… Tab A (T355Y)

Signed-off-by: Siddharth Manthan <siddharth_manthan@outlook.com>

### CODE REVIEW

**Comment by hello-world-321** [comment]
Wiki page for working things and bugs

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+#include "msm8916-pm8916.dtsi"
+#include "msm8916-no-psci.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/sound/apq8016-lpass.h>
+
+/ {
+	model = "Samsung Galaxy Tab A 8.0 (2015)";
+	compatible = "samsung,gt58lte-aus", "qcom,msm8916";
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:14]: > Also I think -aus has no difference from other ones except the firmware so you'll drop the suffix in the dt (will still handle firmware in pmOS but devices can share same dt like a3, a3u)

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+		interrupt-parent = <&msmgpio>;
+		interrupts = <115 IRQ_TYPE_EDGE_RISING>;
+		interrupt-names = "INT1";
+
+		st,drdy-int-pin = <1>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&accel_int_default>;
+		mount-matrix = "0", "1", "0",
+					"-1", "0", "0",
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:215]: Indention looks funny, Is your editor configured for 8-wide tab?

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+					"0", "0", "1";
+	};
+
+	light-sensor@10 {
+		status = "disabled"; //Display bug when adjusting brightness
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:220]: Drop the node if it doesn't work/causes problems now. We'll re-add it when it works then

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+
+&blsp_i2c5 {
+	status = "okay";
+
+	/*
+	 * Actual IC is Zinitix bt532.
+	 * It seems compatible enough with bt541 for basic functionality.
+	 */
+
+	//FIXME: Needs driver modification to use touchkeys
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:237]: Comments like this don't belong to the DT.

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts [hunk 5]
```diff
@@ -0,0 +1,643 @@
...
+		pinctrl-0 = <&tsp_int_default>;
+	};
+};
+
+&blsp1_uart2 {
+	status = "okay";
+};
+
+&dsi0 {
+	//FIXME: Display artifacts when changing brightness.
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:265]: > Comments like this don't belong to the DT.

(this is a known bug in the mdss for a lot of people apparently)

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts [hunk 6]
```diff
@@ -0,0 +1,643 @@
...
+};
+
+&sound {
+	status = "okay";
+
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&cdc_pdm_lines_act &ext_sec_tlmm_lines_act &secondary_mic>;
+	pinctrl-1 = <&cdc_pdm_lines_sus &ext_sec_tlmm_lines_sus &secondary_mic>;
+
+	model = "samsung-a2015";
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:342]: This probably needs to be changed (cc @Minecrell)

Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:342]: It depends if it needs to be handled differently from samsung-a2015. gt58lte also has tfa9895. But I guess it doesn't have SecondaryMic or earpiece maybe?

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts (hunk 3 above)
Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:220]: Well, this suggests that the sensor itself is working fine, just the auto brightness is problematic. In may be better to disable the automatic brightness then?

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts (hunk 6 above)
Comment (hello-world-321) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:342]: It has both secondary mic and earpiece

Comment (hello-world-321) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:342]: Other than having the secondary mic on gpio, it is same as samsung-a2015.

Comment (hello-world-321) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:342]: Other than having the secondary mic on gpio, it is same as samsung-a2015.

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts (hunk 2 above)
Comment (hello-world-321) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-gt58lte.dts:215]: No it was 4. Converted both files.

**Comment by hello-world-321** [comment]
Requires enabling the following kernel drivers :
1.  CM3323
2. INPUT_GPIO_VIBRA

**Comment by TravMurav** [comment]
(now duplicated )

**Pull request closed without merging**